pkg/network/vmispec/restart_test.go

```diff
@@ -0,0 +1,197 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package vmispec_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+var _ = Describe("Restart", func() {
+	const (
+		secondaryNetName1 = "foo"
+		secondaryNADName1 = "foo-nad"
+
+		secondaryNetName2 = "bar"
+		secondaryNADName2 = "bar-nad"
+	)
+
+	It("should not require restart the VM when there are no networks", func() {
+		vmi := libvmi.New(
+			libvmi.WithAutoAttachPodInterface(false),
+		)
+		vm := libvmi.NewVirtualMachine(vmi)
+
+		Expect(vmispec.ShouldChangeRequireRestart(vm, vmi)).To(BeFalse())
+	})
+
+	It("should not require restart the VM when there are no changes", func() {
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(secondaryNetName1)),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			libvmi.WithNetwork(libvmi.MultusNetwork(secondaryNetName1, secondaryNADName1)),
+		)
+		vm := libvmi.NewVirtualMachine(vmi).DeepCopy()
+
+		Expect(vmispec.ShouldChangeRequireRestart(vm, vmi)).To(BeFalse())
+	})
+
+	It("should not require restart the VM when networks are added", func() {
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		)
+		vm := libvmi.NewVirtualMachine(vmi).DeepCopy()
```

Comment:
Something for a follow up but I wonder if we should always `DeepCopy()` before returning the vm from `NewVirtualMachine`?

Reply:
It makes sense to me.